vault/resource_kv_secret_v2.go

```diff
@@ -184,13 +261,72 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 				if err := d.Set(consts.FieldMetadata, serializeDataMapToString(v)); err != nil {
 					return diag.FromErr(err)
 				}
+
+				// Read & Set custom metadata
+				if _, ok := v[consts.FieldCustomMetadata]; ok {
+					cm, err := readKVV2Metadata(d, client)
+					if err != nil {
+						return diag.FromErr(err)
+					}
+
+					customMetadata := []interface{}{cm}
+					if err := d.Set(consts.FieldCustomMetadata, customMetadata); err != nil {
+						return diag.FromErr(err)
+					}
+				}
 			}
 		}
+
 	}
 
 	return nil
 }
 
+func readKVV2Metadata(d *schema.ResourceData, client *api.Client) (map[string]interface{}, error) {
+	mount := d.Get(consts.FieldMount).(string)
```

Comment:
probably want to bail out early if `d.Id()` is `""` ? An alternative to that, would be to change the signature to take the `mount` and `name` instead of `d`.